core/account.py

```diff
@@ -122,10 +119,7 @@ def put_account(account_id: str) -> Tuple[dict, int]:
             db.session.query(Account)
             .filter(
                 Account.id == account_id,
-                or_(
-                    Account.azure_ad_subject_id == azure_ad_subject_id,
-                    Account.azure_ad_subject_id.is_(None),
-                ),
+                Account.azure_ad_subject_id == azure_ad_subject_id,
```

Comment:
I'm not sure this is logically correct?

Previously, if you did a `put_account` with an `account_id` and a non-None `azure_ad_subject_id`, it would update the account even if in the DB there was a null `azure_ad_subject_id`.

Now, it will throw an error because it can't find an account. Is that intentional/correct?

Reply:
Before it would update an Account if it had a matching `account_id` and (matching `azure_ad_subject_id` OR the `azure_ad_subject_id` is `None`).

I am defaulting `azure_ad_subject_id` to `None` if none was provided. So the `==` condition _should_ always work. Or so I thought..

Reply:
Changed it back to what it was.

Reply:
Maybe we should add a test for this? 😆

Reply:
Added.